### DISCUSSION TITLE
Docker Windows / Container Monitoring

### DISCUSSION THREAD

**Discussion by jaffro2k** [jasonacox/Powerwall-Dashboard#403] [Q&A]
Hi all - I'm wondering what folks use for monitoring their actual containers and the Docker runtime?  As I'm on Docker Desktop for Windows it does crash every now and then and would like to be able to send alerts via email (and also view performance stats etc).  Any thoughts appreciated (and yes I should probably move the containers to something more reliable :-).

**Comment by jasonacox** [comment]
That's interesting! Docker should restart any crashed container and that is my experience for the last several years on Linux, RPi and MacOS. I just started testing on Win11 but haven't experienced crashes. Perhaps some of our Windows gurus can help?

**Reply by jaffro2k** [reply]
Thanks @jasonacox - yes it does happen rarely but Docker itself just crashes and doesn't re-start (unless I restart Windows).

**Comment by BJReplay** [comment]
I'm running on WSL.  I used to be running on Docker Desktop for Windows against the WSL backend, but I'm now running on docker just running on WSL (which is not 100-level, but 200-level - it is harder to get going, and doesn't have the GUI level monitoring).

The issue on Windows is that you rely on Docker Desktop to keep your docker environment running.  Even on WSL, it can be more flakey - but that's why I moved away from docker desktop.  I'm not about to suggest that as an alternative, because it is much harder; I'd suggest RPi instead, TBH.

If the docker desktop process stop, it tends to take docker with it, and with it, your Powerwall-Dashboard.

So, as @jaffro2k identified in the OP, the issue is something more reliable; not so much the underlying OS, but Docker Desktop.

If you're game, you can try following these instructions https://github.com/jasonacox/Powerwall-Dashboard/blob/main/WINDOWS.md to get docker running on WSL2 on Windows 11 - it is very reliable for me, but is a bit more work to get set up.  Once set up, remember, don't log out of your WSL terminal session - just close it.  Logging out will result in the WSL underlying session closing, and that will tear down docker as well, due to the way WSL works.  These instructions rely on you starting a background process `tmux new-session -d -s keepalive` to "trick" WSL in keeping it's process running, which allows docker to keep running.

**Comment by BJReplay** [comment]
> Docker itself just crashes and doesn't re-start (unless I restart Windows).

That's exactly what happens on Windows.  Docker Desktop isn't meant for production workloads, and Powerwall Dashboard is (more or less) a production workload :)

**Comment by ThePnuts** [comment]
Could you potentially write a script that fires off ever 5 min or so that checks if the docker service is running and if not, runs the application again?
